code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -179,6 +180,12 @@ var/list/gaslist_cache = null
 					//Prevents whatever mechanism is causing it to hit negative temperatures.
 				//world << "post [temperature], [cached_gases["plasma"][MOLES]], [cached_gases["co2"][MOLES]]
 			*/
+	if(cached_gases["freon"])
+		if(cached_gases["freon"][MOLES] > MOLES_PLASMA_VISIBLE)
+			if(return_temperature() < 50)
+				if(holder)

```

Comment:
I'd do this check first.
it's 1 atom var lookup vs. 2-3 var lookups+assoc lookups + 2 value comparisons.
the ```if(holder)``` is definitely the cheaper to check continuously.